In [1]:
import pandas as pd
import numpy as np

In [2]:
pipeline_df = pd.read_csv("Data/pipeline.csv")

In [3]:
df_frost = pipeline_df.loc[(pipeline_df["Acc_City"] == "FROST") & (pipeline_df['Acc_State'] == "TX")]

In [4]:
df_frost["Acc_Long"]

1012   -96.86338
1502    96.86614
Name: Acc_Long, dtype: float64

### Frost, TX is supposed to be a negative longitude so amending index [1502] to be -96.86614

In [5]:
df_linden = pipeline_df.loc[(pipeline_df["Acc_City"] == "LINDEN") & (pipeline_df['Acc_State'] == "NJ")]

In [6]:
df_linden["Acc_Long"]

42     -74.239900
112    -74.238300
127    -74.217550
320    -74.237250
339    -74.219917
366    -74.258350
403    -74.209062
822    -74.239999
948    -74.221761
1000   -74.208518
1032   -74.222200
1343   -74.239070
1371   -74.211753
1470   -74.239568
1514   -74.239141
1766   -74.239401
1815   -74.241097
1994   -74.240044
2032   -74.239114
2193   -14.355830
2446   -74.240350
2539   -74.235210
2698   -74.240296
Name: Acc_Long, dtype: float64

### Index[2193] needs to change from -14.355830 to -74.355830

In [7]:
df_mobile = pipeline_df.loc[(pipeline_df["Acc_City"] == "MOBILE") & (pipeline_df['Acc_State'] == "AL")]

In [8]:
df_mobile["Acc_Long"]

1555    88.173136
Name: Acc_Long, dtype: float64

### Index[1555] needs to be negative.  Change to -88.173136

In [9]:
df_yoder = pipeline_df.loc[(pipeline_df["Acc_City"] == "YODER") & (pipeline_df['Acc_State'] == "WY")]

In [10]:
df_yoder["Acc_Long"]

1755    104.263399
Name: Acc_Long, dtype: float64

### Index[1755] needs to be negative.  Change to -104.263399

### Making amendments to latitudes and longitudes

In [11]:
pipeline_df = pd.read_csv("Data/pipeline.csv")

### FROST, TX

In [12]:
n=pipeline_df['Acc_Long'][1502]

In [13]:
a=(n * -1)

In [14]:
pipeline_df['Acc_Long'][1502]=a

C:\Users\USER\anaconda3\envs\PythonData\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


### LINDEN, NJ

In [15]:
pipeline_df['Acc_Long'][2193]=-74.355830

C:\Users\USER\anaconda3\envs\PythonData\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


### MOBILE, AL

In [16]:
n=pipeline_df['Acc_Long'][1555]

In [17]:
a=(n * -1)

In [18]:
pipeline_df['Acc_Long'][1555]=a

C:\Users\USER\anaconda3\envs\PythonData\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


### YODER, WY

In [19]:
n=pipeline_df['Acc_Long'][1755]

In [20]:
a=(n * -1)

In [21]:
pipeline_df['Acc_Long'][1755]=a

C:\Users\USER\anaconda3\envs\PythonData\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [24]:
pipeline_df.to_csv(r'C:\Users\USER\Documents\Analysis_Projects\Oil_Pipelines\Oil_Pipelines\Data\pipeline3.csv', index = False, header=True)